Update bootstrap script

andreas-bok-sociomantic opened the pull request.

gautam-kotian-sociomantic commented on bootstrap.sh:
```diff
@@ -305,6 +307,8 @@ function set_VENDOR {
         AUTORECONF_REBUILD_HOST=true
       elif [[ "x$VENDOR_RELEASE" == 'xquantal' ]]; then
         AUTORECONF_REBUILD_HOST=true
+      elif [[ "x$VENDOR_RELEASE" == 'xtrusty' ]]; then
```
Is it expected that `raring` doesn't appear between `quantal` & `trsusty`?
Because it does appear in lines 229-230 above.

andreas-bok-sociomantic replied:
Ah. Missed that. Will add it

gautam-kotian-sociomantic commented on bootstrap.sh:
```diff
@@ -1653,7 +1657,8 @@ function main {
   #                  RHEL{rhel,Tikanga,Santiago}
-  #                  Ubuntu{ubuntu,Lucid,Maverick,Natty,Oneiric,Precise,Quantal}
+  #                  Ubuntu{ubuntu,Lucid,Maverick,Natty,Oneiric,Precise,Quantal,
+  #                         Trusty}
```
Here again `Raring` is missing.

gautam-kotian-sociomantic commented on bootstrap.sh:
```diff
@@ -145,8 +145,8 @@ function rebuild_host_os {
 
-#  values: darwin,fedora,rhel,ubuntu,debian,opensuse
 # Validate the distribution name, or toss an error
+# values: darwin,fedora,rhel,ubuntu,debian,opensuse
```
Since you're changing this line anyway, it'd be nicer to add a space following the commas.

ben-palmer-sociomantic commented on bootstrap.sh:
```diff
@@ -1653,7 +1659,8 @@ function main {
   #                  RHEL{rhel,Tikanga,Santiago}
-  #                  Ubuntu{ubuntu,Lucid,Maverick,Natty,Oneiric,Precise,Quantal}
+  #                  Ubuntu{ubuntu,Lucid,Maverick,Natty,Oneiric,Precise,Quantal,
+  #                         Raring,Trusty,Xenial}
```
If we are adding Xenial in this commit should we also add it to the `$VENDOR_RELEASE` changes?

ben-palmer-sociomantic commented on bootstrap.sh:
```diff
@@ -146,7 +146,7 @@ function rebuild_host_os {
 
 # Validate the distribution name, or toss an error
-# values: darwin,fedora,rhel,ubuntu,debian,opensuse
+# values: darwin,fedora,rhel,ubuntu,debian,suse,opensuse
```
I would combine this commit with the first one.

ben-palmer-sociomantic commented on tests/unit/datetypes.c:
```diff
@@ -89,7 +89,19 @@ int main(int argc, char *argv[])
+      "b DATE,"
+      "c YEAR(4),"
+      "d TIMESTAMP(0),"
+      "e TIMESTAMP(6) DEFAULT CURRENT_TIMESTAMP(6),"
```
I might be wrong but wasn't this change already merged in another PR?

andreas-bok-sociomantic replied:
Yes, but onto branch v5.4.x. So that commit shouldn't go into this PR.

ben-palmer-sociomantic reviewed (commented):
The changes LGTM I just had a couple of comments about the organisation of the commits.

andreas-bok-sociomantic committed:
Fixes for vendor names and release checks

- Add missing vendors and releases to documentation
- Add cases to switch for missing releases for osx and ubuntu
- Minor formatting changes for enumerations and switch/case

andreas-bok-sociomantic committed:
Fix typo

andreas-bok-sociomantic commented:
- Removed commit from v5.4.x branch
- Fixed wrong osx release names
- Squashed commits

ben-palmer-sociomantic commented on bootstrap.sh:
```diff
@@ -305,6 +304,13 @@ function set_VENDOR {
+        AUTORECONF_REBUILD_HOST=true
+      elif [[ "x$VENDOR_RELEASE" == 'xtrusty' ]]; then
+        AUTORECONF_REBUILD_HOST=true
+      fi
```
Is this `fi` supposed to be here?

andreas-bok-sociomantic replied:
nope, shouldn't

ben-palmer-sociomantic commented:
One comment then LGTM.

andreas-bok-sociomantic commented:
- Removed wrong `fi` stmt

```diff 
diff --git a/bootstrap.sh b/bootstrap.sh
index 9b9f98a..4afc75a 100755
--- a/bootstrap.sh
+++ b/bootstrap.sh
@@ -308,7 +308,6 @@ function set_VENDOR {
         AUTORECONF_REBUILD_HOST=true
       elif [[ "x$VENDOR_RELEASE" == 'xtrusty' ]]; then
         AUTORECONF_REBUILD_HOST=true
-      fi
       elif [[ "x$VENDOR_RELEASE" == 'xxenial' ]]; then
         AUTORECONF_REBUILD_HOST=true
       fi
```

ben-palmer-sociomantic commented:
LGTM
